core/src/main/java/org/bitcoinj/evolution/SimplifiedMasternodeListManager.java

```diff
@@ -394,8 +402,15 @@ public void close() {
                 quorumRotationState.removeEventListeners(blockChain, peerGroup);
             }
 
+            try {
+                threadPool.shutdown();
+                threadPool.awaitTermination(Long.MAX_VALUE, TimeUnit.SECONDS);
+            } catch (InterruptedException x) {
+                // swallow
+            }
```

Comment:
If there is a job in progress, it will finish, but if there are more than one job the other jobs will be cancelled.